src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/DictionaryDefaultConverter.cs

```diff
@@ -69,9 +69,7 @@ protected static JsonConverter<TValue> GetValueConverter(ref WriteStack state)
             ref ReadStack state,
             [MaybeNullWhen(false)] out TCollection value)
         {
-            bool shouldReadPreservedReferences = options.ReferenceHandling.ShouldReadPreservedReferences();
-
-            if (!state.SupportContinuation && !shouldReadPreservedReferences)
+            if (!state.SupportContinuation && options.ReferenceHandler == null)
```

Comment:
It looks like these 2 variables are always checked together. Although they could be separate in the future, consider collapsing then having `state.SupportContinuation` consider `options.ReferencHandler` factored in to prevent the &&

Reply:
As of today, `state.SupportContinuation` and `options.ReferenceHandler` need to go through the "slow path" since both relies on the `state.Current.ObjectState`. 

Is your suggestion related to refactor the code so we can do a fast path for `ReferenceHandler` when there is no need for continuation? If that's the case I can try to tackle that in a follow-up PR.

Reply:
Not a fast path. Just having a single bool like
`state.UseFastPath = !state.SupportContinuation && options.ReferenceHandler == null`